### DISCUSSION TITLE
Issue with Mover Package Implementation in FloPy: Model Error and Manual Fix Required

### DISCUSSION THREAD

**Discussion by YozeGholami** [modflowpy/flopy#2313] [General]
Hi
I am working with a model that contains two lakes, and I want to use the outlet of Lake 1 to flow into Lake 2. To achieve this, I applied the following code for the package data and period data of the Mover package (as shown in Figure 1 in the attachment). This code aligns with FloPy's requirements for defining the Mover package. The code successfully created a recarray for the package data and period data (Figure 2 in the attachment).

I verified this setup in the model's output file (Figure 3 in the attachment), and it matches the specifications of MODFLOW 6 according to the MODFLOW 6 input/output documentation. However, when I ran the model, I encountered an error (Figure 4 in the attachment).

To resolve this issue, I manually modified the input file (Figure 5 in the attachment), after which the model ran successfully. My question is: why did my initial attempt to build the model using FloPy result in an error, and what might have caused the issue with the generated code?
Fig1:
![Fig1](https://github.com/user-attachments/assets/5523a19d-e8ce-4496-8256-4dd1a5661801)

Fig2:
![Fig2](https://github.com/user-attachments/assets/2af5d5bc-29cf-4312-869a-a564eb137d18)

Fig3:
![Fig3](https://github.com/user-attachments/assets/673c74ac-d237-4930-9adc-d9ecc759bf1d)

Fig4:
![Fig4](https://github.com/user-attachments/assets/192e827d-18e5-400a-9fc1-4e8ceee4286c)

Fig5:
![Fig5](https://github.com/user-attachments/assets/d7cd63b3-ea38-4656-af72-42c07e4102d3)

**Comment by hxfan1227** [comment]
I'm collaborating with @YozeGholami on the same project. To clarify, we have divided the entire lake into two separate parts (i.e., lak1 and lak2), each with a defined outlet. We're using the Mover package to transfer water from the lak1 outlet to lak2. This approach aims to simulate the real hydrological process of the entire lake, where the stage difference between the two parts of the lake is approximately 0.8 meters. Any suggestions are welcome on whether this approach is adaptable or if more reliable methods exist.

**Comment by jlarsen-usgs** [comment]
I think you may have to add `mover=True` to each of your lake packages.

**Reply by YozeGholami** [reply]
Thanks for the suggestion. I've already added mover=True  the lake package. I currently have one lake package that includes four lakes. Perhaps I should define more lake packages and assign each lake to its own package. I’ll test this approach.

Best regards,
Amin
